tests/ui/pinned_drop-1.rs

```diff
@@ -0,0 +1,17 @@
+use pin_project_lite::pin_project;
+
+pin_project! {
+    pub struct S {
+        #[pin]
+        field: u8,
+    }
+    impl PinnedDrop for S {
+        fn drop(self: Pin<&mut Self>) {
+            self.__drop_inner();
+        }
+    }
+}
```

Comment:
Hmm..., I expected this to fail to compile, but it seems that it actually compiles.

Reply:
(resolved by using normal arg  instead `self`)